### PR TITLE
Add integration tests for group local and cloud snaps

### PR DESCRIPTION
**Pull request by harsh-px**
Signed-off-by: Harsh Desai <harsh@portworx.com>

### CODE REVIEW

**Comment by harsh-px** [comment]
PR for torpedo changes: https://github.com/portworx/torpedo/pull/229

File: vendor/github.com/portworx/torpedo/drivers/scheduler/k8s/k8s.go [hunk 1]
```diff
@@ -215,6 +215,8 @@ func validateSpec(in interface{}) (interface{}, error) {
 		return specObj, nil
 	} else if specObj, ok := in.(*snap_v1.VolumeSnapshot); ok {
 		return specObj, nil
+	} else if specObj, ok := in.(*stork_api.GroupVolumeSnapshot); ok {
```
Comment (disrani-px) [inline vendor/github.com/portworx/torpedo/drivers/scheduler/k8s/k8s.go]: Was this updated with dep? I don't see the Gopkg.* updates

File: test/integration_test/specs/mysql-3d-snap-crd/3d-snapshot.yaml [hunk 2]
```diff
@@ -1,9 +1,9 @@
 apiVersion: stork.libopenstorage.org/v1alpha1
 kind: GroupVolumeSnapshot
 metadata:
-  name: mysql-groupsnapshot
+  name: mysql-3d-snapshot
```
Comment (disrani-px) [inline test/integration_test/specs/mysql-3d-snap-crd/3d-snapshot.yaml]: Don't use 3d-snapshot, use groupsnapshot

File: test/integration_test/specs/mysql-3d-snap-crd/3d-snapshot.yaml [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
 spec:
+  preSnapshotRule: px-presnap-rule
```
Comment (disrani-px) [inline test/integration_test/specs/mysql-3d-snap-crd/3d-snapshot.yaml]: `preExecRule`

File: test/integration_test/specs/mysql-3d-snap-crd/px-presnap-rule.yaml [hunk 4]
```diff
@@ -1,7 +1,9 @@
 apiVersion: stork.libopenstorage.org/v1alpha1
 kind: Rule
 metadata:
-  name: mysql-presnap-rule
+  name: px-presnap-rule
```
Comment (disrani-px) [inline test/integration_test/specs/mysql-3d-snap-crd/px-presnap-rule.yaml]: Undo these changes

File: test/integration_test/specs/mysql-snap-group-fail/group-snapshot-fail-noexists-rule.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+apiVersion: stork.libopenstorage.org/v1alpha1
+kind: GroupVolumeSnapshot
+metadata:
+  name: mysql-groupsnapshot-noexist-rule
+spec:
+  preSnapshotRule: nosuchrule
```
Comment (disrani-px) [inline test/integration_test/specs/mysql-snap-group-fail/group-snapshot-fail-noexists-rule.yaml]: `preExecRule`

File: vendor/github.com/portworx/torpedo/drivers/scheduler/k8s/k8s.go (hunk 1 above)
Comment (disrani-px) [reply vendor/github.com/portworx/torpedo/drivers/scheduler/k8s/k8s.go]: Updated vendor

**Comment by disrani-px** [comment]
Fixed review comments
